instrumentation/netty/netty-3.8/src/main/java/io/opentelemetry/instrumentation/auto/netty/v3_8/client/NettyHttpClientTracer.java

```diff
@@ -22,19 +22,34 @@
 import static org.jboss.netty.handler.codec.http.HttpHeaders.Names.HOST;
 
 import io.grpc.Context;
-import io.opentelemetry.OpenTelemetry;
 import io.opentelemetry.context.Scope;
 import io.opentelemetry.context.propagation.HttpTextFormat.Setter;
 import io.opentelemetry.instrumentation.api.tracer.HttpClientTracer;
 import io.opentelemetry.trace.Span;
 import java.net.URI;
 import java.net.URISyntaxException;
+import org.jboss.netty.handler.codec.http.HttpHeaders;
 import org.jboss.netty.handler.codec.http.HttpRequest;
 import org.jboss.netty.handler.codec.http.HttpResponse;
 
-public class NettyHttpClientTracer extends HttpClientTracer<HttpRequest, HttpResponse> {
+public class NettyHttpClientTracer
+    extends HttpClientTracer<HttpRequest, HttpHeaders, HttpResponse> {
   public static final NettyHttpClientTracer TRACER = new NettyHttpClientTracer();
 
+  @Override
+  public Scope startScope(Span span, HttpHeaders headers) {
+    if (!headers.contains("amz-sdk-invocation-id")) {
+      return super.startScope(span, headers);
+    } else {
+      // TODO (trask) if we move injection up to aws-sdk layer, and start suppressing nested netty
```

Comment:
Hmm I remember hoping to have removed this check but bringing it back but not much else. Will check it out to try to remember